src/Surfnet/ServiceProviderDashboard/Application/Service/ServiceConnectionService.php

```diff
@@ -86,55 +91,52 @@ private function listTestIdps()
         return $this->testIdps->createCollection()->testEntities();
     }
 
-    private function addIdpList(EntityConnectionCollection $collection): void
-    {
-        $collection->addIdpList($this->listTestIdps());
-    }
-
     /**
      * @param array<string, IdentityProvider> $testIdpsIndexed
+     * @param array<string, IdentityProvider> $otherIdpsIndexed
      */
     private function addEntitiesToCollection(
         EntityConnectionCollection $collection,
         InstitutionId $institutionId,
         array $testIdpsIndexed,
-        ?Service $service = null,
+        array $otherIdpsIndexed,
     ): void {
         $list = [];
         $entities = $this->entityService->findPublishedTestEntitiesByInstitutionId($institutionId);
+
         if ($entities === null) {
             $collection->addEntityConnections($list);
             return;
         }
         $allowedProtocols = [Constants::TYPE_SAML, Constants::TYPE_OPENID_CONNECT_TNG];
         foreach ($entities as $entity) {
-            if (!in_array($entity->getProtocol()->getProtocol(), $allowedProtocols)) {
-                // Skipping irrelevant entity types
+            $metadata = $this->assertValidMetadata($entity);
+            $team = $this->getTeam($metadata);
+            $service = $this->serviceService->getServiceByTeamName($team);
+            if ($service === null || !in_array($entity->getProtocol()->getProtocol(), $allowedProtocols)) {
```

Comment:
Making a lokal method like `isKnownService` then you can make it clearer and remove the comment perhaps